source/server/server.cc

```diff
@@ -205,7 +199,7 @@ void InstanceImpl::initialize(Options& options, TestHooks& hooks,
   // GuardDog (deadlock detection) object and thread setup before workers are
   // started and before our own run() loop runs.
   guard_dog_.reset(
-      new Server::GuardDogImpl(*admin_scope_, *config_, ProdMonotonicTimeSource::instance_));
+      new Server::GuardDogImpl(stats_store_, *config_, ProdMonotonicTimeSource::instance_));
```

Comment:
Apologies this was just miscommunication. I meant that GuardDog should operate on a Stats::Scope vs. a Stats::Store. Stats::Store derives from Stats::Scope, and that's obviously different from what the scope actually is. In this case we effectively want the root scope since these are root/global stats.